tests/unit_tests/test_wave_06.py

```diff
@@ -227,11 +253,82 @@ def test_swap_best_by_category_no_other_match_is_false():
         their_priority="Decor"
     )
 
-    raise Exception("Complete this test according to comments below.")
+    assert not result
+    assert len(tai.inventory) == 3
+    assert len(jesse.inventory) == 3
+    assert item_a in tai.inventory
+    assert item_b in tai.inventory
+    assert item_c in tai.inventory
+    assert item_d in jesse.inventory
+    assert item_e in jesse.inventory
+    assert item_f in jesse.inventory
     # *********************************************************************
     # ****** Complete Assert Portion of this test **********
     # *********************************************************************
     # Assertions should check:
     # - That result is falsy
     # - That tai and jesse's inventories are the correct length
     # - That all the correct items are in tai and jesse's inventories
+
+
+
+
+# unit tests for optional part of project
+
+def test_get_by_age_default_age():
+    electronic = Electronics()
+    assert electronic.age == None
+
+def test_get_by_age():
+    cloth = Clothing(condition = 2, age=1)
+    assert cloth.age == 1
+
+
+def test_newest_item_return_false_for_empty_inventory():
+    tai = Vendor(
+        inventory=[])
+    assert tai.newest_item() == None
+
+
+def test_newest_item():
+    item_a = Decor(age=5.0)
+    item_b = Electronics(age=4.0)
+    item_c = Decor(age=3.0)
+    item_d = Clothing(age=9.0)
+    item_e = Decor(age=4.0)
+    item_f = Clothing(age=2.0)
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c, item_d, item_e, item_f]
+    )  
+    new_item = tai.newest_item()
+    
+    assert new_item ==  item_f  
+        
+def test_swap_by_newest():
+    item_a = Decor(age=1.0)
+    item_b = Electronics(age=4.0)
+    item_c = Decor(age=3.0)
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c]
+    )
+
+    item_d = Clothing(age=2.0)
+    item_e = Decor(age=4.0)
+    item_f = Clothing(age=7.0)
+    jesse = Vendor(
+        inventory=[item_d, item_e, item_f]
+    )
+
+    result = tai.swap_by_newest(vendor_friend=jesse)
+
+    assert result
+    assert len(tai.inventory) == 3
+    assert len(jesse.inventory) == 3
+    assert item_a not in tai.inventory
+    assert item_b in tai.inventory
+    assert item_c in tai.inventory
+    assert item_d in tai.inventory
+    assert item_d not in jesse.inventory
+    assert item_e in jesse.inventory
+    assert item_f in jesse.inventory
```

Comment:
Great comprehensive tests of the successful swap, length of inventories, and individual items within the inventories!